src/useSnapEdgeToParent/useSnapEdgeToParent.ts

```diff
@@ -1,28 +1,25 @@
-import { useRef, useState, useMemo, useLayoutEffect } from 'react'
-import { Positioning, getPosition, AutoPosition, BreakDistance, Refs } from './helpers'
+import { useState, useMemo, useLayoutEffect } from 'react'
+import { Positioning, getPosition, AutoPosition, BreakDistance, RefAndRectMap, Vertical, Horizontal } from './helpers'
+import { useClientRect }from '../useClientRect'
 
-export function useSnapEdgeToParent<
-  Target extends HTMLElement = HTMLDivElement,
-  Parent extends HTMLElement = HTMLDivElement,
-  Container extends HTMLElement = HTMLDivElement
-> (
-  breakDistances: BreakDistance | number,
-  initialLayout?: AutoPosition
-): [Refs<Target, Parent, Container>, Positioning, () => void] {
+export function useSnapEdgeToParent (breakDistances: BreakDistance | number, initialPositioning?: AutoPosition) {
+  const [targetRect, targetRef] = useClientRect()
+  const [parentRect, parentRef] = useClientRect()
+  const [containerRect, containerRef] = useClientRect()
 
-  const refs = {
-    target: useRef<Target>(null),
-    parent: useRef<Parent>(null),
-    container: useRef<Container>(null)
+  const refAndRectMap = {
+    target: { rect: targetRect, ref: targetRef },
+    parent: { rect: parentRect, ref: parentRef },
+    container: { rect: containerRect, ref: containerRef }
   }
 
-  const [layout, setLayout] = useState<Positioning>({ horizontal: 'left', vertical: 'top' })
+  const [positioning, setPositioning] = useState<Positioning>({ horizontal: Horizontal.RIGHT, vertical: Vertical.BOTTOM })
 
-  const handleLayout = () => setLayout(getPosition(refs, breakDistances, initialLayout))
+  const handlePositioning = () => setPositioning(getPosition(refAndRectMap, breakDistances, initialPositioning))
```

Comment:
`getPositioning` would be more consistent and accurate here than `getPosition`

Reply:
I think getPosition means the current position and it is good, but recalculatePositioning I agree that is a good name instead handlePositioning function

Reply:
done